File: apps/api/app/schemas/membership.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class ActivateMembershipRequest(BaseModel):
    phone: str = Field(min_length=9, max_length=15)


class PaymentStatusResponse(BaseModel):
    id: uuid.UUID
    status: str
    amount: float
    phone: str
    mpesa_receipt: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class MembershipStatusResponse(BaseModel):
    membership_status: str
    latest_payment: PaymentStatusResponse | None

File: apps/web/src/lib/api/membership.ts
import { apiFetch } from "./core";
import type { PaymentStatus } from "./shared";

export type Payment = {
  id: string;
  status: PaymentStatus;
  amount: number;
  phone: string;
  mpesa_receipt: string | null;
  created_at: string;
};

export type MembershipStatusResponse = {
  membership_status: string;
  latest_payment: Payment | null;
};

export const membershipApi = {
  activate: (phone: string) =>
    apiFetch<Payment>("/membership/activate", { method: "POST", body: JSON.stringify({ phone }) }),
  status: () => apiFetch<MembershipStatusResponse>("/membership/status"),
};
